### ISSUE TITLE
Design deployment workflow for the site

### ISSUE THREAD

**Issue by hagbardc** [IndivisibleOnline/Indivisible-Plugin#10]
With multiple developers for the site, and a site that is live (and hopefully under more regular use) we should have a deployment workflow that allows us to deploy and rollback packages safely and confidently.  Features of such a deployment workflow should include:

- Ability to work on our local machines
- Get a local instance of the site up and running in just a few steps
- Provide documentation that allows someone to understand the code / site structure and plugin dependencies relatively easily
- 1st stage of deployment should be to a (the) staging site.  Deployments should be handled via code checkout from a specified branch of the github repo (we should not be pushing code from local machines direct to stage or prod)
- 2nd stage of deployment should be to the prod site, again direct from the github repo

**Comment by ksnyder-mtm** [comment]
@darvelo -- Would you own this task? This is super important, and we are planning to have an in-person meeting the week of June 10. Would you be able to provide recommendations for this?

**Comment by imlindaryan** [comment]
I recommend this be part of an in-person Tech team conversation.  When can this happen?  @howardstevens @darvelo @kennybatf

**Comment by darvelo** [comment]
@iwksnyder @imlindaryan Yes, I'll own this task. I'm available any time for an in-person Tech team meeting! It would be good if we could have at least a few hours to build something solid together.

I recommend using [Docker](https://www.docker.com/) for building the virtual machine environment and using Git wherever possible to track code changes. Docker is easy to install.

I've got the public-facing site mostly working when running locally within a Docker container, a virtual machine that can be destroyed and rebuilt at will with the guarantee that the system files are identical on each run. I'm currently working on figuring out how to:

1. Get login working
1. Include git-tracked projects (like this repo) within the container
2. Ensure changes within the Wordpress admin are tracked so that we know exactly what changes are in a build. Wordpress makes changes directly to the database so this will be a challenge.
